docs/development/data-flow/README.md

```diff
@@ -0,0 +1,98 @@
+# Data flow and Usage
+
+Regardless of the type of installation of OpenProject, the following diagram provides a high-level overview of through which systems data related to OpenProject is flowing.
+
+
+
+```mermaid
+%%{init: {'theme':'neutral'}}%%
+
+flowchart TD
+  A[Client Browser] -->|"HTTP(s) requests"| B(Load Balancer / Proxy)
+  A1[API or Native clients] -->|"HTTP(s) requests"| B(Load Balancer / Proxy)
+  B -->|Proxy| openproject
+
+  subgraph openproject[OpenProject]
+    C[Puma app server]
+    D[Background worker]
+  end
+    
+
+  subgraph integrations[External integrations]
+	  O[Other integrations]
+    N[Nextcloud]
+  end
+  
+  subgraph services[Services]
+  	M[memcached]
+	  P[PostgreSQL]
+	  S[Object storage or NFS]
+  end
+  
+
+  openproject <--> services
+  openproject --> integrations
+  B <--> integrations
```

Comment:
There are cases where an S3 or equivalent is used. This could also only be mentioned in the text as it might otherwise dilute the digram's clarity. On the other hand, I am ok with not having cloudfront mentioned as that is specific for our SaaS installation. 

In case of SVN or GIT being hosted, that would be another client.

Reply:
I'll add object storage as a separate node, as that might be relevant for on-premises as well. SVN and Git access OpenProject only through HTTPs, so they're effectively another client. I can add them though for completeness.